Libraries/src/net/relinc/libraries/application/Bar.java

```diff
@@ -77,6 +91,30 @@ public double getRadius() {
 		return this.diameter / 2;
 	}
 
+	public void parseJSONtoParameters(String input) {
+		JSONObject jsonObject = null;
+		JSONParser jsonParser = new JSONParser();
+
+		try {
+			jsonObject = (JSONObject) jsonParser.parse(input);
+		} catch (org.json.simple.parser.ParseException e) {
+			//TODO throw e
+		}
+
+		setParametersJSON(jsonObject);
+	}
+
+	private void setParametersJSON(JSONObject jsonObject) {
+		name = (String)jsonObject.get(nameDescrip);
```

Comment:
casting is generally not sweet but this looks like how the library is supposed to be used